enterprise/internal/insights/compression/commits.go

```diff
@@ -0,0 +1,157 @@
+package compression
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/keegancsmith/sqlf"
+
+	"github.com/sourcegraph/sourcegraph/internal/api"
+	"github.com/sourcegraph/sourcegraph/internal/database/basestore"
+	"github.com/sourcegraph/sourcegraph/internal/database/dbutil"
+	"github.com/sourcegraph/sourcegraph/internal/vcs/git"
+)
+
+type DBCommitStore struct {
+	*basestore.Store
+}
+
+type CommitStore interface {
+	Save(ctx context.Context, id api.RepoID, commit *git.Commit) error
+	Get(ctx context.Context, id api.RepoID, start time.Time, end time.Time) ([]CommitStamp, error)
+	GetMetadata(ctx context.Context, id api.RepoID) (CommitIndexMetadata, error)
+	UpsertMetadataStamp(ctx context.Context, id api.RepoID) (CommitIndexMetadata, error)
+	InsertCommits(ctx context.Context, id api.RepoID, commits []*git.Commit) error
+}
+
+func NewCommitStore(db dbutil.DB) *DBCommitStore {
+	return &DBCommitStore{
+		Store: basestore.NewWithDB(db, sql.TxOptions{}),
+	}
+}
+
+func (c *DBCommitStore) With(other basestore.ShareableStore) *DBCommitStore {
+	return &DBCommitStore{Store: c.Store.With(other)}
+}
+
+func (c *DBCommitStore) Transact(ctx context.Context) (*DBCommitStore, error) {
+	txBase, err := c.Store.Transact(ctx)
+	return &DBCommitStore{Store: txBase}, err
+}
+
+func (c *DBCommitStore) Save(ctx context.Context, id api.RepoID, commit *git.Commit) error {
+	ref := commit.ID
+	err := c.Exec(ctx, sqlf.Sprintf(insertCommitIndexStr, id, ref, commit.Committer.Date))
+	if err != nil {
+		return fmt.Errorf("error saving commit for repo_id: %v ref %v: %w", id, ref, err)
```

Comment:
```suggestion
		return fmt.Errorf("error saving commit for repo_id: %v commit_id %v: %w", id, ref, err)
```

Usually in Sourcegraph's code there are three ways of describing a commit:

* _a commit ID_ - always the 40-character SHA you're used to
* _a branch name_
* `rev`: Any git revision: a commit ID, short commit ID, branch name, relative spec like `HEAD~12`, etc.

based on that, I think you should call this `commit_id` instead of `ref`

Reply:
X-ref: Going with `commit` for consistency with other tables